moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -963,21 +963,29 @@ bool TimeOptimalTrajectoryGeneration::computeTimeStamps(robot_trajectory::RobotT
   {
     moveit::core::RobotStatePtr waypoint = trajectory.getWayPointPtr(p);
     Eigen::VectorXd new_point(num_joints);
+    double duration = trajectory.getWayPointDurationFromPrevious(p);
     // The first point should always be kept
     bool diverse_point = (p == 0);
 
     for (size_t j = 0; j < num_joints; ++j)
     {
       new_point[j] = waypoint->getVariablePosition(idx[j]);
-      // If any joint angle is different, it's a unique waypoint
-      if (p > 0 && std::fabs(new_point[j] - points.back()[j]) > min_angle_change_)
+      // If any joint angle is different and duration is nonzero, it's a unique waypoint
```

Comment:
@AndyZe can you explain how this is supposed to fix the test case? All waypoints have a nonzero timestamp so this change shouldn't affect the outcome. Also, TOTG should not consider existing timestamps as it's performing a complete resampling and recomputing of timestamps, it doesn't take existing times into account.

Reply:
OK, that was a mistake. Will revert.

Please test locally, though. It works well for myself and one other.

Reply:
I think I added this comment about duration, but it should be removed.

`// If any joint angle is different, it's a unique waypoint`